diff --git a/node_modules/@credo-ts/core/build/agent/EnvelopeService.js b/node_modules/@credo-ts/core/build/agent/EnvelopeService.js
index 72821a7..d11d209 100644
--- a/node_modules/@credo-ts/core/build/agent/EnvelopeService.js
+++ b/node_modules/@credo-ts/core/build/agent/EnvelopeService.js
@@ -38,6 +38,7 @@ let EnvelopeService = class EnvelopeService {
                 message: encryptedMessage,
             });
             recipientKeysBase58 = [routingKeyBase58];
+            forwardMessage["messageType"] = message['@type'];
             this.logger.debug('Forward message created', forwardMessage);
             const forwardJson = forwardMessage.toJSON({
                 useDidSovPrefixWhereAllowed: agentContext.config.useDidSovPrefixWhereAllowed,
diff --git a/node_modules/@credo-ts/core/build/modules/routing/messages/ForwardMessage.d.ts b/node_modules/@credo-ts/core/build/modules/routing/messages/ForwardMessage.d.ts
index 4f8577b..396f78a 100644
--- a/node_modules/@credo-ts/core/build/modules/routing/messages/ForwardMessage.d.ts
+++ b/node_modules/@credo-ts/core/build/modules/routing/messages/ForwardMessage.d.ts
@@ -3,6 +3,7 @@ import { EncryptedMessage } from '../../../types';
 export interface ForwardMessageOptions {
     id?: string;
     to: string;
+    messageType: string;
     message: EncryptedMessage;
 }
 /**
@@ -19,5 +20,6 @@ export declare class ForwardMessage extends AgentMessage {
     readonly type: string;
     static readonly type: import("../../../utils/messageType").ParsedMessageType;
     to: string;
+    messageType: string;
     message: EncryptedMessage;
 }
diff --git a/node_modules/@credo-ts/core/build/types.d.ts b/node_modules/@credo-ts/core/build/types.d.ts
index 0d52d00..4bdb736 100644
--- a/node_modules/@credo-ts/core/build/types.d.ts
+++ b/node_modules/@credo-ts/core/build/types.d.ts
@@ -82,6 +82,7 @@ export interface PlaintextMessage {
         thid?: string;
         pthid?: string;
     };
+     messageType: string;
     [key: string]: unknown;
 }
 export interface OutboundPackage {
