diff --git a/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js b/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
index fb1fb9d..260fe21 100644
--- a/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
+++ b/node_modules/@credo-ts/core/build/modules/credentials/protocol/v2/V2CredentialProtocol.js
@@ -444,6 +444,7 @@ class V2CredentialProtocol extends BaseCredentialProtocol_1.BaseCredentialProtoc
         let credentialRecord = await this.findByProperties(messageContext.agentContext, {
             threadId: requestMessage.threadId,
             role: models_1.CredentialRole.Issuer,
+            connectionId: connection?.id
         });
         const formatServices = this.getFormatServicesFromMessage(requestMessage.formats);
         if (formatServices.length === 0) {
