diff --git a/node_modules/@credo-ts/core/build/modules/vc/W3cCredentialServiceOptions.d.ts b/node_modules/@credo-ts/core/build/modules/vc/W3cCredentialServiceOptions.d.ts
index 43f5d7b..6e5cbbc 100644
--- a/node_modules/@credo-ts/core/build/modules/vc/W3cCredentialServiceOptions.d.ts
+++ b/node_modules/@credo-ts/core/build/modules/vc/W3cCredentialServiceOptions.d.ts
@@ -7,6 +7,7 @@ import type { W3cCredential } from './models/credential/W3cCredential';
 import type { W3cPresentation } from './models/presentation/W3cPresentation';
 import type { JwaSignatureAlgorithm } from '../../crypto/jose/jwa';
 import type { SingleOrArray } from '../../utils/type';
+import { Credential } from '@hyperledger/anoncreds-nodejs';
 export type W3cSignCredentialOptions<Format extends ClaimFormat.JwtVc | ClaimFormat.LdpVc | undefined = undefined> = Format extends ClaimFormat.JwtVc ? W3cJwtSignCredentialOptions : Format extends ClaimFormat.LdpVc ? W3cJsonLdSignCredentialOptions : W3cJwtSignCredentialOptions | W3cJsonLdSignCredentialOptions;
 export type W3cVerifyCredentialOptions<Format extends ClaimFormat.JwtVc | ClaimFormat.LdpVc | undefined = undefined> = Format extends ClaimFormat.JwtVc ? W3cJwtVerifyCredentialOptions : Format extends ClaimFormat.LdpVc ? W3cJsonLdVerifyCredentialOptions : W3cJwtVerifyCredentialOptions | W3cJsonLdVerifyCredentialOptions;
 export type W3cSignPresentationOptions<Format extends ClaimFormat.JwtVp | ClaimFormat.LdpVp | undefined = undefined> = Format extends ClaimFormat.JwtVp ? W3cJwtSignPresentationOptions : Format extends ClaimFormat.LdpVp ? W3cJsonLdSignPresentationOptions : W3cJwtSignPresentationOptions | W3cJsonLdSignPresentationOptions;
@@ -21,7 +22,7 @@ interface W3cSignCredentialOptionsBase {
     /**
      * The credential to be signed.
      */
-    credential: W3cCredential;
+    credential: any;
     /**
      * URI of the verificationMethod to be used for signing the credential.
      *
