diff --git a/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApi.js b/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApi.js
index 7b54cfb..3ed891d 100644
--- a/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApi.js
+++ b/node_modules/@credo-ts/core/build/modules/credentials/CredentialsApi.js
@@ -415,10 +415,17 @@ let CredentialsApi = class CredentialsApi {
         return credentialRecord;
     }
     async getFormatData(credentialRecordId) {
-        const credentialRecord = await this.getById(credentialRecordId);
-        const protocol = this.getProtocol(credentialRecord.protocolVersion);
-        return protocol.getFormatData(this.agentContext, credentialRecordId);
+        try {
+            const credentialRecord = await this.getById(credentialRecordId);
+            const protocol = this.getProtocol(credentialRecord.protocolVersion);
+            return protocol.getFormatData(this.agentContext, credentialRecordId);
+        } catch (error) {
+            this.logger?.error?.(`Error in getFormatData for credential ID ${credentialRecordId}: ${error?.message}`, error);
+            return null; // Prevents application from crashing
+        }
     }
+
+
     /**
      * Retrieve a credential record by id
      *
@@ -500,11 +507,11 @@ exports.CredentialsApi = CredentialsApi = __decorate([
     (0, plugins_1.injectable)(),
     __param(3, (0, plugins_1.inject)(constants_1.InjectionSymbols.Logger)),
     __metadata("design:paramtypes", [MessageSender_1.MessageSender,
-        services_1.ConnectionService,
-        agent_1.AgentContext, Object, CredentialRepository_1.CredentialRepository,
-        RoutingService_1.RoutingService,
-        DidCommMessageRepository_1.DidCommMessageRepository,
-        services_2.RevocationNotificationService,
-        CredentialsModuleConfig_1.CredentialsModuleConfig])
+    services_1.ConnectionService,
+    agent_1.AgentContext, Object, CredentialRepository_1.CredentialRepository,
+    RoutingService_1.RoutingService,
+    DidCommMessageRepository_1.DidCommMessageRepository,
+    services_2.RevocationNotificationService,
+    CredentialsModuleConfig_1.CredentialsModuleConfig])
 ], CredentialsApi);
 //# sourceMappingURL=CredentialsApi.js.map
\ No newline at end of file
