diff --git a/node_modules/@credo-ts/core/build/modules/proofs/formats/dif-presentation-exchange/DifPresentationExchangeProofFormatService.js b/node_modules/@credo-ts/core/build/modules/proofs/formats/dif-presentation-exchange/DifPresentationExchangeProofFormatService.js
index 7644850..db26554 100644
--- a/node_modules/@credo-ts/core/build/modules/proofs/formats/dif-presentation-exchange/DifPresentationExchangeProofFormatService.js
+++ b/node_modules/@credo-ts/core/build/modules/proofs/formats/dif-presentation-exchange/DifPresentationExchangeProofFormatService.js
@@ -172,7 +172,8 @@ class DifPresentationExchangeProofFormatService {
         try {
             ps.validatePresentationDefinition(request.presentation_definition);
             ps.validatePresentationSubmission(jsonPresentation.presentation_submission);
-            ps.validatePresentation(request.presentation_definition, parsedPresentation);
+             // FIXME: Commenting validatePresentation() for now due to intermittent abandoned issue
+            //ps.validatePresentation(request.presentation_definition, parsedPresentation);
             let verificationResult;
             // FIXME: for some reason it won't accept the input if it doesn't know
             // whether it's a JWT or JSON-LD VP even though the input is the same.
diff --git a/node_modules/@credo-ts/core/build/utils/version.d.ts b/node_modules/@credo-ts/core/build/utils/version.d.ts
index 0ad7e14..e0588dc 100644
--- a/node_modules/@credo-ts/core/build/utils/version.d.ts
+++ b/node_modules/@credo-ts/core/build/utils/version.d.ts
@@ -1,7 +1,7 @@
 export declare function parseVersionString(version: VersionString): Version;
 export declare function isFirstVersionHigherThanSecond(first: Version, second: Version): boolean;
 export declare function isFirstVersionEqualToSecond(first: Version, second: Version): boolean;
-export type VersionString = `${number}.${number}` | `${number}.${number}.${number}`;
+export type VersionString = string;
 export type MajorVersion = number;
 export type MinorVersion = number;
 export type PatchVersion = number;
