diff --git a/node_modules/@digitalcredentials/jsonld/lib/context.js b/node_modules/@digitalcredentials/jsonld/lib/context.js
index 5f0de78..1315061 100644
--- a/node_modules/@digitalcredentials/jsonld/lib/context.js
+++ b/node_modules/@digitalcredentials/jsonld/lib/context.js
@@ -920,6 +920,11 @@ api.createTermDefinition = ({
     if(!_deepCompare(previousMapping, mapping)) {
       const protectedMode = (options && options.protectedMode) || 'error';
       if(protectedMode === 'error') {
+        if (term === 'VerifiableCredential' || term === 'VerifiablePresentation') {
+          console.warn("Term is 'VerifiableCredential'. Logging a warning instead of throwing an error.");
+          console.warn('WARNING: protected term redefinition', { term });
+          return;
+        }
         throw new JsonLdError(
           `Invalid JSON-LD syntax; tried to redefine "${term}" which is a ` +
           'protected term.',
